libs/remote_storage/tests/test_real_s3.rs

```diff
@@ -259,6 +259,44 @@ async fn s3_upload_download_works(ctx: &mut MaybeEnabledS3) -> anyhow::Result<()
     Ok(())
 }
 
+#[test_context(MaybeEnabledS3)]
+#[tokio::test]
+async fn s3_copy_works(ctx: &mut MaybeEnabledS3) -> anyhow::Result<()> {
```

Comment:
nit: this test could be included in https://github.com/neondatabase/neon/pull/6155

Reply:
Yeah I made a bunch of PRs to the same crate and they conflict a little bit with each other. I will rebase them on each other in the order I am merging stuff.